### PR TITLE
(I#3934) Update user manual and in-app documentation for splitByLength()

### LINKED ISSUE #3934: Update user manual and in-app documentation for splitByLength()
The documentation, specifically for splitByLength() does not inform the user that they have the possibility to only return values within a given range of indices, Iv since updated the document but wanted to make a bug report on it for submitting the PR.

ex: 
`"123456789012345678901234567890".splitByLengths(3,4,5)[0,2]`
`returns [ "123",  "4567" ]`

More about this issue can be seen in this thread: https://github.com/OpenRefine/OpenRefine/issues/3929

### PR DESCRIPTION
**Pull request by TheTrailblazer09**
Fixes #3934

Changes proposed in this pull request:
 - updated user manual and documentation for splitByLength()

### CODE REVIEW

**Comment by antoine2711** [comment]
> Fixes #{3934}

Hi @TheTrailblazer09 ! Just some quick advise for your description: the first line must start with:
"Fixes" + SPACE + "#" + Issue number. No curly braces like you added. 

Regards, Antoine

**Comment by TheTrailblazer09** [comment]
> > Fixes #~{~ 3934 ~}~
> 
> Hi @TheTrailblazer09 ! Just some quick advise for your description: the first line must start with: "Fixes" + SPACE + "#" + Issue number. No curly braces like you added.
> 
> When you are ready, tell me, I'll review your changes.
> 
> Regards, Antoine

Hi, I am ready

**Comment by TheTrailblazer09** [comment]
> > Fixes #~{~ 3934 ~}~
> 
> Hi @TheTrailblazer09 ! Just some quick advise for your description: the first line must start with: "Fixes" + SPACE + "#" + Issue number. No curly braces like you added.
> 
> When you are ready, tell me, I'll review your changes.
> 
> Regards, Antoine

Hi, I am ready

**Review by wetneb — COMMENTED** [review]
I think I disagree with the original issue actually: the array slicing has nothing to do with the function call: all arrays can be sliced in this way, so I would not add it to this section of the documentation.

I guess we could mention that in some cases it might be useful to chain this function call with a slicing operator, but it should really be clear slicing is available in other contexts, regardless of which function produced a given array. So I would leave the section header untouched, at least.

**Comment by TheTrailblazer09** [comment]
> I think I disagree with the original issue actually: the array slicing has nothing to do with the function call: all arrays can be sliced in this way, so I would not add it to this section of the documentation.
> 
> I guess we could mention that in some cases it might be useful to chain this function call with a slicing operator, but it should really be clear slicing is available in other contexts, regardless of which function produced a given array. So I would leave the section header untouched, at least.

So do you think I should change the documentation (to say that slicing is available in other contexts as well ) or should I let it be?

File: docs/docs/manual/grelfunctions.md [hunk 1]
```diff
@@ -174,9 +174,19 @@ Returns a string converted to a number. Will attempt to convert other formats in
 
 Returns the array of strings obtained by splitting s by sep. The separator can be either a string or a regex pattern. For example, `"fire, water, earth, air".split(",")` returns an array of 4 strings: [ "fire", " water", " earth", " air" ]. Note that the space characters are retained but the separator is removed. If you include “true” for the preserveTokens boolean, empty segments are preserved.
 
-###### splitByLengths(s, n1, n2, ...) {#splitbylengthss-n1-n2-}
+###### splitByLengths(s, n1, n2, ...) [(optional) i1, i2] {#splitbylengthss-n1-n2-}
```
Comment (antoine2711) [inline docs/docs/manual/grelfunctions.md:177]:
```suggestion
###### splitByLengths(s, n1, n2, ...) {#splitbylengthss-n1-n2-}
```
This should be there. It has nothing specific about this function.

For the dash at the end of the hash, could you change it? It's going to require changes elsewhere...

File: docs/docs/manual/grelfunctions.md [hunk 2]
```diff
@@ -174,9 +174,19 @@ Returns a string converted to a number. Will attempt to convert other formats in
...
 
-Returns the array of strings obtained by splitting s into substrings with the given lengths. For example, `"internationalization".splitByLengths(5, 6, 3)` returns an array of 3 strings: [ "inter", "nation", "ali" ]. Excess characters are discarded.
+Returns the array of strings obtained by splitting s into substrings with the given lengths. For example, `"internationalization".splitByLengths(5, 6, 3)` returns an array of 3 strings: [ "inter", "nation", "ali" ]. Excess characters are discarded from the output array.
+
+Optionally, it also allows array slicing on the returned array. In this case, it returns the array consisting of a subset of elements between i1 and (i2 – 1)
```
Comment (antoine2711) [inline docs/docs/manual/grelfunctions.md:181]:
```suggestion
Like other functions that returns an array, it also allows array slicing on the returned array. In that case, it returns the array consisting of a subset of elements between i1 and (i2 – 1)
```
@wetneb: what do you think?

Comment (wetneb) [reply docs/docs/manual/grelfunctions.md]: Yes, better! Just "Like other functions that return", not "returns"

File: main/src/com/google/refine/expr/functions/strings/SplitByLengths.java [hunk 3]
```diff
@@ -69,7 +69,7 @@ public Object call(Properties bindings, Object[] args) {
...
+        return "Returns the array of strings obtained by splitting s into substrings with the given lengths. For example, \"internationalization\".splitByLengths(5, 6, 3) returns an array of 3 strings: [ \"inter\", \"nation\", \"ali\" ]. Excess characters are discarded from the output array. \nLike other functions that returns an array, it also allows array slicing on the returned array. In that case, it returns the array consisting of a subset of elements between i1 and (i2 – 1). For example, (a) \"internationalization\".splitByLengths(5, 6, 3)[0,3] returns an array of 3 strings: [\"inter\", \"nation\", \"ali\"] \n (b) \"internationalization\".splitByLengths(5, 6, 3)[0,2] returns an array of 2 strings: [\"inter\", \"nation\"] \n (c) \"internationalization\".splitByLengths(5, 6, 3)[1,3] returns an array of 2 strings: [\"nation\", \"ali\" ] \n (d) \"internationalization\".splitByLengths(5, 6, 3)[1] returns string at position 1: \"nation\"";
```
Comment (antoine2711) [inline main/src/com/google/refine/expr/functions/strings/SplitByLengths.java:72]:
```suggestion
        return "Returns the array of strings obtained by splitting s into substrings with the given lengths. For example, \"internationalization\".splitByLengths(5, 6, 3) returns an array of 3 strings: [ \"inter\", \"nation\", \"ali\" ]. Excess characters are discarded from the output array.
```

This is way too much, and not specific, and in this small window, it's important to stay brief and on target. I wouldn't add this:
`Like other functions that returns an array, it also allows array slicing on the returned array. In that case, it returns the array consisting of a subset of elements between i1 and (i2 – 1). For example, (a) \"internationalization\".splitByLengths(5, 6, 3)[0,3] returns an array of 3 strings: [\"inter\", \"nation\", \"ali\"] \n (b) \"internationalization\".splitByLengths(5, 6, 3)[0,2] returns an array of 2 strings: [\"inter\", \"nation\"] \n (c) \"internationalization\".splitByLengths(5, 6, 3)[1,3] returns an array of 2 strings: [\"nation\", \"ali\" ] \n (d) \"internationalization\".splitByLengths(5, 6, 3)[1] returns string at position 1: \"nation\"";`

**Review by antoine2711 — CHANGES_REQUESTED** [review]
There is a problem in `main/src/com/google/refine/expr/functions/strings/SplitByLengths.java`

Regards, Antoine

File: docs/docs/manual/grelfunctions.md (hunk 1 above)
Comment (wetneb) [reply docs/docs/manual/grelfunctions.md]: I'd say it's fine to leave the hash as it is…

Comment (antoine2711) [reply docs/docs/manual/grelfunctions.md]: > I'd say it's fine to leave the hash as it is…

What's the logic @wetneb?
- [ ] It's autogenerated, and as such, it's extra work to tweak it.
- [ ] It's there to show that other param can optionnally come.
- [ ] Just not worth it, user doesn't really see it.

Regards, Antoine

**Review by wetneb — APPROVED** [review]

File: docs/versioned_docs/version-3.5/manual/grelfunctions.md [hunk 4]
```diff
@@ -174,9 +174,19 @@ Returns a string converted to a number. Will attempt to convert other formats in
 
 Returns the array of strings obtained by splitting s by sep. The separator can be either a string or a regex pattern. For example, `"fire, water, earth, air".split(",")` returns an array of 4 strings: [ "fire", " water", " earth", " air" ]. Note that the space characters are retained but the separator is removed. If you include “true” for the preserveTokens boolean, empty segments are preserved.
 
-###### splitByLengths(s, n1, n2, ...) {#splitbylengthss-n1-n2-}
+###### splitByLengths(s, n1, n2, ...){#splitbylengthss-n1-n2}
```
Comment (antoine2711) [inline docs/versioned_docs/version-3.5/manual/grelfunctions.md:177]:
```suggestion
###### splitByLengths(s, n1, n2, ...){#splitbylengthss-n1-n2-}
```

**Review by antoine2711 — APPROVED** [review]
Good work. Thank you, @TheTrailblazer09.

**Comment by TheTrailblazer09** [comment]
> Good work. Thank you, @TheTrailblazer09.

Thanks & Warm Regards

**Pull request merged**